packages/costa/src/runtime.ts

```diff
@@ -552,6 +584,11 @@ export class CostaRuntime {
         // TODO(feely): not implement the version
         installDir, `node_modules/${pkg.name}@${pkg.version}`)
     };
+    if (source && source.from !== 'npm') {
```

Comment:
Source check and assignments should be moved forward the fetch* methods.

Reply:
done